File: src/programmers/internal/json_schema_description.ts
import { IJsDocTagInfo } from "../../schemas/metadata/IJsDocTagInfo";

export const json_schema_description = (props: {
  description?: string | null | undefined;
  jsDocTags?: IJsDocTagInfo[];
}): string | undefined =>
  props.jsDocTags
    ?.find((tag) => tag.name === "description")
    ?.text?.[0]?.text?.split("\r\n")
    .join("\n") ??
  props.description ??
  undefined;

File: packages/typia/native/core/programmers/iterate/json_schema_description.go
package iterate

import (
	"strings"

	nativemetadata "github.com/samchon/typia/packages/typia/native/core/schemas/metadata"
)

func json_schema_description(props struct {
	description *string
	jsDocTags   []nativemetadata.IJsDocTagInfo
}) *string {
	for _, tag := range props.jsDocTags {
		if tag.Name != "description" || len(tag.Text) == 0 {
			continue
		}
		text := strings.ReplaceAll(tag.Text[0].Text, "\r\n", "\n")
		return &text
	}
	return props.description
}
